tools/building.py

```diff
@@ -1224,18 +1224,43 @@ def run_wasm_opt(infile, outfile=None, args=[], **kwargs):  # noqa
   return run_binaryen_command('wasm-opt', infile, outfile, args=args, **kwargs)
 
 
-def save_intermediate(src, dst):
+def save_intermediate(src, name):
   if DEBUG:
-    dst = 'emcc-%02d-%s' % (save_intermediate.counter, dst)
+    dst = 'emcc-%02d-%s' % (save_intermediate.counter, name)
     save_intermediate.counter += 1
     dst = os.path.join(shared.CANONICAL_TEMP_DIR, dst)
-    logger.debug('saving debug copy %s' % dst)
+    logger.debug('saving intermediate file %s' % dst)
     shutil.copyfile(src, dst)
 
 
+def write_intermediate(content, name):
+  if DEBUG:
+    dst = 'emcc-%02d-%s' % (save_intermediate.counter, name)
+    save_intermediate.counter += 1
+    dst = os.path.join(shared.CANONICAL_TEMP_DIR, dst)
+    logger.debug('saving intermediate file %s' % dst)
+    utils.write_file(dst, content)
```

Comment:
```suggestion
  if DEBUG:
    utils.write_file(dst, content)
    save_intermediate(content, name)
```
Can this not work, to reuse the code?

Reply:
Oh, I guess it generates the name there... but perhaps another way to share the code is to pass `content` as an optional param to `save_intermediate`?

Reply:
Refactored.